server/data/hmppsAuthClient.test.ts

```diff
@@ -14,8 +14,16 @@ jest.mock('redis', () => ({
   set: jest.fn().mockImplementation((key, value, command, ttl, callback) => callback(null, null)),
 }))
 
+interface MockRedis {
+  on: jest.Mock
+  get: jest.Mock
+  set: jest.Mock
+}
+
+const mockRedis = (redis as unknown) as MockRedis
```

Comment:
how come we need the intermediate cast to `unknown` here?

Reply:
Without it you get this error message from the compiler:

```
    server/data/hmppsAuthClient.test.ts:23:19 - error TS2352: Conversion of type 'typeof import("/Users/lawrence/code/interventions/hmpps-interventions-ui/node_modules/@types/redis/index")' to type 'MockRedis' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
      Type 'typeof import("/Users/lawrence/code/interventions/hmpps-interventions-ui/node_modules/@types/redis/index")' is missing the following properties from type 'MockRedis': on, get, set
```

I think it's because there's no way for the compiler to know that we've dynamically added the `on` / `get` / `set` methods in the `jest.mock` call, but I'm not sure exactly why that means the `unknown` is needed.

Reply:
(This whole mock is a bit weird-feeling though, but I couldn't find a better way of doing it using the stuff that Jest / ts-jest gives you. Probably a limitation of my JS / TS knowledge though so might be able to revisit it once I'm a bit more clued up.)

Reply:
👍 thanks for the explanation!